src/Handler/StreamHandler.php

```diff
@@ -286,7 +288,7 @@ private function createStream(RequestInterface $request, array $options)
             foreach ($options as $key => $value) {
                 $method = "add_{$key}";
                 if (isset($methods[$method])) {
-                    $this->{$method}($request, $context, $value, $params);
+                    $this->{$method}($request, $context, $value, $params, $uri);
```

Comment:
Is it really necessary to add the URI as a parameter? It's accessible from the request, isn't it?